### PR TITLE
dclock: init at 0.1.0

### PR DESCRIPTION
**Pull request by travgm**
Adding the dclock package to nixpkgs and updated the maintainers file as I will maintain the package. 

## Things done

- Built on platform(s)
  - [X ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/misc/dclock/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{ lib, stdenv, fetchFromGitHub, haskellPackages }:
```
Comment (genga898) [inline pkgs/applications/misc/dclock/default.nix:1]: Kindly use [nixfmt](https://search.nixos.org/packages?channel=24.11&from=0&size=50&sort=relevance&type=packages&query=nixfmt) to format your package

File: pkgs/applications/misc/dclock/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+with lib;
```
Comment (genga898) [inline pkgs/applications/misc/dclock/default.nix:3]:
```suggestion

```

File: pkgs/applications/misc/dclock/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    lens
+    machines
+    optparse-applicative
+    text
+    time
+    process
+  ];
+
+  description =
+    "A decimal clock that breaks your day into a 1000 decimal minutes";
```
Comment (genga898) [inline pkgs/applications/misc/dclock/default.nix:36]:
```suggestion
    "Decimal clock that breaks your day into a 1000 decimal minutes";
```

**Review by genga898 — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/dclock/default.nix (hunk 3 above)
Comment (travgm) [reply pkgs/applications/misc/dclock/default.nix]: Updated description

File: pkgs/applications/misc/dclock/default.nix (hunk 2 above)
Comment (travgm) [reply pkgs/applications/misc/dclock/default.nix]: Removed in most recent update

File: pkgs/applications/misc/dclock/default.nix (hunk 1 above)
Comment (travgm) [reply pkgs/applications/misc/dclock/default.nix]: Ran on most recent update

**Comment by genga898** [comment]
A few things to change about your commits, when adding a new maintainer your commit message should follow this format `add: maintainer travgm` and your commit for adding the package should be of the format `package_name: init at version_number`

You only need the two commits for the maintainers list and the package, the rest of the commits can be squashed through `git rebase`

**Comment by travgm** [comment]
> A few things to change about your commits, when adding a new maintainer your commit message should follow this format `add: maintainer travgm` and your commit for adding the package should be of the format `package_name: init at version_number`
> 
> You only need the two commits for the maintainers list and the package, the rest of the commits can be squashed through `git rebase`
> 
> You can reference my [first commit](https://github.com/NixOS/nixpkgs/pull/347300#event-14616729669) here and see how I was guided

Thanks, I believe I got it updated now

**Comment by genga898** [comment]
You weren't supposed to squash the add maintainers commit, it is seperate from the init  one

**Comment by travgm** [comment]
> You weren't supposed to squash the add maintainers commit, it is seperate from the init one

SHould be fixed now, my apologies.

**Comment by genga898** [comment]
> > You weren't supposed to squash the add maintainers commit, it is seperate from the init one
> 
> Should be fixed now, my apologies.

Apologies, made an error with the maintainers commit, it's supposed to take the format as update in the above comment

**Comment by genga898** [comment]
You can now edit the title of the pr to match your commit for package initialization.

**Comment by travgm** [comment]
> You can now edit the title of the pr to match your commit for package initialization.

Updated!

**Comment by genga898** [comment]
When you make a new change, it is best to squash the commit into the init one, it makes things a bit easier, also all commits follow the same pattern of `package_name: fix invalid path`

**Review by genga898 — CHANGES_REQUESTED** [review]
```nix
description = "Decimal clock that breaks your day into a 1000 decimal minutes";
  homepage = "https://github.com/travgm/dclock";
  license = lib.licenses.mit;
  maintainers = with lib.maintainers; [ travgm ];
  mainProgram = "dclock";
```

It should be wrapped under the meta attribute 
```nix
meta = {
      description = "Decimal clock that breaks your day into a 1000 decimal minutes";
      homepage = "https://github.com/travgm/dclock";
      license = lib.licenses.mit;
      maintainers = with lib.maintainers; [ travgm ];
      mainProgram = "dclock";
}
```

File: pkgs/by-name/dc/dclock/package.nix [hunk 4]
```diff
@@ -35,11 +35,11 @@ haskellPackages.mkDerivation {
     process
   ];
 
-  meta = {
+  meta = with lib; {
```
Comment (genga898) [inline pkgs/by-name/dc/dclock/package.nix:38]:
```suggestion
  meta = {
```
It's mostly advised you use `with lib;` only in places it is required like license and maintainers.

File: pkgs/by-name/dc/dclock/package.nix [hunk 5]
```diff
@@ -35,11 +35,11 @@ haskellPackages.mkDerivation {
...
     description = "Decimal clock that breaks your day into a 1000 decimal minutes";
     homepage = "https://github.com/travgm/dclock";
     license = lib.licenses.mit;
-    maintainers = with lib.maintainers; [ travgm ];
+    maintainers = with maintainers; [ travgm ];
```
Comment (genga898) [inline pkgs/by-name/dc/dclock/package.nix:42]:
```suggestion
    maintainers = with lib.maintainers; [ travgm ];
```

**Review by genga898 — CHANGES_REQUESTED** [review]

File: pkgs/by-name/dc/dclock/package.nix (hunk 5 above)
Comment (travgm) [reply pkgs/by-name/dc/dclock/package.nix:42]: Not sure why it is still failing, every example and package I have checked has it this way as of my last update.

Comment (travgm) [reply pkgs/by-name/dc/dclock/package.nix]: according to this for haskell packages how it is now is correct https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/haskell.section.md#chap-meta

**Comment by travgm** [comment]
> ```nix
> description = "Decimal clock that breaks your day into a 1000 decimal minutes";
>   homepage = "https://github.com/travgm/dclock";
>   license = lib.licenses.mit;
>   maintainers = with lib.maintainers; [ travgm ];
>   mainProgram = "dclock";
> ```
> 
> It should be wrapped under the meta attribute 
> ```nix
> meta = {
>       description = "Decimal clock that breaks your day into a 1000 decimal minutes";
>       homepage = "https://github.com/travgm/dclock";
>       license = lib.licenses.mit;
>       maintainers = with lib.maintainers; [ travgm ];
>       mainProgram = "dclock";
> }
> ```

This is resolved as the haskell package is different. The build is now successfully passing.

**Pull request merged**